src/anomalib/models/__init__.py

```diff
@@ -98,39 +84,71 @@ def get_available_models() -> list[str]:
         >>> get_available_models()
         ['ai_vad', 'cfa', 'cflow', 'csflow', 'dfkde', 'dfm', 'draem', 'efficient_ad', 'fastflow', ...]
     """
-    return [convert_pascal_to_snake_case(cls.__name__) for cls in AnomalyModule.__subclasses__()]
-
+    return [_convert_pascal_to_snake_case(cls.__name__) for cls in AnomalyModule.__subclasses__()]
 
-def get_model(config: DictConfig | ListConfig) -> AnomalyModule:
-    """Load model from the configuration file.
 
-    Works only when the convention for model naming is followed.
-
-    The convention for writing model classes is
-    `anomalib.models.<model_name>.lightning_model.<ModelName>Lightning`
-    `anomalib.models.stfpm.lightning_model.StfpmLightning`
+def _get_model_by_name(name: str) -> AnomalyModule:
+    """Get's the model by name.
 
     Args:
-        config (DictConfig | ListConfig): Config.yaml loaded using OmegaConf
+        name (str): Name of the model. The name is case insensitive.
 
     Raises:
-        ValueError: If unsupported model is passed
+        ModuleNotFoundError: If unsupported model is passed
 
     Returns:
         AnomalyModule: Anomaly Model
     """
     logger.info("Loading the model.")
-    model: AnomalyModule
+    model: AnomalyModule | None = None
+
+    # search for model by name in available models.
+    for model_name in get_available_models():
+        if name.lower() in model_name:
+            module = import_module(f"anomalib.models.{model_name}")
+            for class_name in dir(module):
+                if class_name.lower() in class_name.lower():
+                    model_class = getattr(module, class_name, None)
+                    if inspect.isclass(model_class):
+                        model = model_class()
+                        break
+            break
+    if model is None:
+        logger.exception(f"Could not find the model {name}. Available models are {get_available_models()}")
+        raise UnknownModelError
+
+    return model
+
+
+def get_model(config: DictConfig | ListConfig | str | Namespace) -> AnomalyModule:
+    """Get Anomaly Model.
+
+    Args:
+        config (DictConfig | ListConfig | str): Can either be a configuration or a string.
 
-    try:
-        module = import_module(".".join(config.model.class_path.split(".")[:-1]))
-        model = getattr(module, config.model.class_path.split(".")[-1])
-        model = model(**config.model.init_args)
-    except ModuleNotFoundError:
-        logger.exception("Could not find the model class: %s", config.model.class_path)
-        raise
+    Examples:
+        >>> get_model("Padim")
```

Comment:
would it be possible to get the model when the first charachter is lower-case such as `padim`?

Reply:
I can convert to title case but this might fail for models like ReverseDistillation, EfficientAd, and such models that have capitalization in the middle

Reply:
For EfficientAd, we already use `efficient_ad` to trigger the efficient ad training. It might be an idea to do the same for `reverse_distillation`?

I'm trying to think about from a user perspective. @djdameln, what are your thoughts on this ?